### PR TITLE
translate rules-essential

### PR DESCRIPTION
**Pull request by babakzy**

### CODE REVIEW

**Review by the-pesar — APPROVED** [review]

File: src/style-guide/rules-essential.md [hunk 1]
```diff
@@ -235,9 +237,9 @@ When Vue processes directives, `v-if` has a higher priority than `v-for`, so tha
 </ul>
 ```
 
-Will throw an error, because the `v-if` directive will be evaluated first and the iteration variable `user` does not exist at this moment.
+باعث ارور میشه, چون دایرکتیو `v-if` اول ارزیابی میشه و متغیر  `user` در اون موقع هنوز وجود نداره.
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:240]:
```suggestion
باعث خطا می‌شود چون دایرکتیو `v-if` اول ارزیابی می‌شود و متغیر  `user` در آن لحظه هنوز وجود ندارد.
```

File: src/style-guide/rules-essential.md [hunk 2]
```diff
@@ -235,9 +237,9 @@ When Vue processes directives, `v-if` has a higher priority than `v-for`, so tha
...
 
-This could be fixed by iterating over a computed property instead, like this:
+این میتونه با استفاده از متغیر computed موقع استفاده حلقه درست بشه.
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:242]:
```suggestion
این مشکل می‌تواند با استفاده از یک computed موقع استفاده حلقه درست شود.
```

File: src/style-guide/rules-essential.md [hunk 3]
```diff
@@ -272,7 +274,7 @@ const activeUsers = computed(() => {
 </ul>
 ```
 
-Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` element:
+به صورت جایگزین, ما میتونیم از تگ `<template>` با `v-for` برای نگهداری المنت `<li>` استفاده کنیم:
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:277]:
```suggestion
به صورت جایگزین، می‌توانیم از تگ `<template>` با `v-for` برای نگهداری المنت `<li>` استفاده کنیم:
```

File: src/style-guide/rules-essential.md [hunk 4]
```diff
@@ -329,24 +331,24 @@ Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` ele
 
 </div>
 
-## Use component-scoped styling {#use-component-scoped-styling}
+## برای استایل دهی از component-scoped استفاده بکنید  {#use-component-scoped-styling}
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:334]:
```suggestion
## برای استایل دهی از component-scoped استفاده کنید {#use-component-scoped-styling}
```

File: src/style-guide/rules-essential.md [hunk 5]
```diff
@@ -329,24 +331,24 @@ Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` ele
...
 
-For applications, styles in a top-level `App` component and in layout components may be global, but all other components should always be scoped.
+برای اپلیکیشن ها , استایل سطح بالا در کامپوننت `App` و کامپوننت های layout ممکنه به صورت سراسری باشه ,ولی برای بقیه کامپوننت ها همیشه باید باید استایل ها به صورت scoped باشه.
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:336]:
```suggestion
برای اپلیکیشن‌ها، استایل سطح بالا در کامپوننت `App` و کامپوننت‌های layout ممکن است به صورت سراسری باشد، ولی برای کامپوننت‌های دیگر همیشه باید استایل‌ها به صورت scoped باشد.
```

File: src/style-guide/rules-essential.md [hunk 6]
```diff
@@ -329,24 +331,24 @@ Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` ele
...
 
-This is only relevant for [Single-File Components](/guide/scaling-up/sfc). It does _not_ require that the [`scoped` attribute](https://vue-loader.vuejs.org/en/features/scoped-css.html) be used. Scoping could be through [CSS modules](https://vue-loader.vuejs.org/en/features/css-modules), a class-based strategy such as [BEM](http://getbem.com/), or another library/convention.
+این مورد فقط مربوط به [Single-File Components](/guide/scaling-up/sfc) میشه. و لازم _نیست_ که [ویژگی `scoped`](https://vue-loader.vuejs.org/en/features/scoped-css.html) استفاده بشه. اسکوپ میتونه با [CSS modules](https://vue-loader.vuejs.org/en/features/css-modules) باشه, و متولوژی بر اساس کلاس مثل [BEM](http://getbem.com/), و یا لایبری ها و یا قرارداد های دیگه.
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:338]:
```suggestion
این تنها در مورد [Single-File Components](/guide/scaling-up/sfc) مرتبط است. این موضوع نیازی به استفاده از [`scoped` attribute](https://vue-loader.vuejs.org/en/features/scoped-css.html) ندارد. این محدود کردن می‌تواند از طریق [ماژول‌های CSS](https://vue-loader.vuejs.org/en/features/css-modules)، یک استراتژی مبتنی بر کلاس مانند [BEM](http://getbem.com/) یا یک کتابخانه/توافق دیگر باشد.
```

File: src/style-guide/rules-essential.md [hunk 7]
```diff
@@ -329,24 +331,24 @@ Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` ele
...
 
-**Component libraries, however, should prefer a class-based strategy instead of using the `scoped` attribute.**
+**با این حال کامپوننت لایبرری ها بهتره که به جای صفت  `scoped` از استایل دهی بر اساس کلاس استفاده کنند.**
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:340]:
```suggestion
**اما کتابخانه‌های کامپوننت، بهتر است به جای استفاده از ویژگی `scoped`، از یک استراتژی مبتنی بر کلاس استفاده کنند.**
```

File: src/style-guide/rules-essential.md [hunk 8]
```diff
@@ -329,24 +331,24 @@ Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` ele
...
 
-This makes overriding internal styles easier, with human-readable class names that don't have too high specificity, but are still very unlikely to result in a conflict.
+این باعث میشه بازنویسی استایل ها با کلاس های با اسامی خوانا که لازم نیست خیلی به خصوص باشه راحتتر باشه، در عین حال احتمالش کمه که باعث تداخل تو نتایج بشه.
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:342]:
```suggestion
این امر انجام تغییر در استایل‌های داخلی را آسان‌تر می‌کند، با نام‌گذاری‌های کلاس قابل فهم برای انسان که لازم نیست خیلی به خصوص باشد، اما هنوز احتمال وقوع تداخل را به شدت کاهش می‌دهند.
```

File: src/style-guide/rules-essential.md [hunk 9]
```diff
@@ -329,24 +331,24 @@ Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` ele
...
 
-::: details Detailed Explanation
-If you are developing a large project, working with other developers, or sometimes include 3rd-party HTML/CSS (e.g. from Auth0), consistent scoping will ensure that your styles only apply to the components they are meant for.
+::: توضیح جزئیات
+ اگه شما در حال توسعه پروژه های با ابعاد بزرگ هستید که با برنامه نویس های دیگه ای روش کار میکنید و یا از HTML/CSS های خارجی (مثل Auth0) استفاده میکنید. استفاده اسکوپ باعث میشه که مطمئن بشید استایل های شما فقط به کامپوننت های مورد نظر اعمال میشن.
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:345]:
```suggestion
::: details توضیحات بیشتر
اگر در حال توسعه یک پروژه بزرگ هستید، با دیگر توسعه‌دهندگان همکاری می‌کنید یا گاهی اوقات کد HTML/CSS شخص ثالث را وارد می‌کنید (مثلاً از Auth0)، استفاده مداوم از اسکوپ اطمینان حاصل می‌کند که استایل‌های شما فقط بر روی کامپوننت‌هایی که برایشان طراحی شده‌اند، اعمال شوند.
```

File: src/style-guide/rules-essential.md [hunk 10]
```diff
@@ -329,24 +331,24 @@ Alternatively, we can use a `<template>` tag with `v-for` to wrap the `<li>` ele
...
+::: توضیح جزئیات
+ اگه شما در حال توسعه پروژه های با ابعاد بزرگ هستید که با برنامه نویس های دیگه ای روش کار میکنید و یا از HTML/CSS های خارجی (مثل Auth0) استفاده میکنید. استفاده اسکوپ باعث میشه که مطمئن بشید استایل های شما فقط به کامپوننت های مورد نظر اعمال میشن. 
 
-Beyond the `scoped` attribute, using unique class names can help ensure that 3rd-party CSS does not apply to your own HTML. For example, many projects use the `button`, `btn`, or `icon` class names, so even if not using a strategy such as BEM, adding an app-specific and/or component-specific prefix (e.g. `ButtonClose-icon`) can provide some protection.
+علاوه بر صفت `scoped`, استفاده از اسم کلاس های منحصر به فرد باعث اطمینان میشه که کلاس های CSS های خارجی روی کامپوننت های شما اثری نمیزان. برای مثال در بیشتر پروژه ها از کلاس های `button`, `btn`, یا `icon` استفاده میشه, پس حتی اگه از استراتژی هایی مثل BEM هم استفاده نکنید, اضاقه کردن پیشوند بخصوص اپلیکیشن و یا کامپوننت  ( `ButtonClose-icon` مثل) میتونه تا حدی باعث محافظت بشه.
```
Comment (mostafa-nematpour) [inline src/style-guide/rules-essential.md:347]:
```suggestion
علاوه بر ویژگی `scoped`، استفاده از نام‌های کلاس منحصر به فرد می‌تواند به اطمینان از اینکه CSS شخص ثالث بر روی HTML شما اعمال نشود، کمک کند. به عنوان مثال، بسیاری از پروژه‌ها از نام‌های کلاسی مانند `button`، `btn` یا `icon` استفاده می‌کنند، پس حتی اگر از استراتژیی مانند BEM استفاده نمی‌کنید، اضافه کردن یک پیشوند خاص برای برنامه یا کامپوننت (مثلاً `ButtonClose-icon`) می‌تواند تا حدی باعث محافظت شود.
```

**Review by mostafa-nematpour — COMMENTED** [review]
Hi, Please look at the review, thank you

**Comment by mostafa-nematpour** [comment]
It seems to be merged :)
@the-pesar

**Pull request merged**